### PR TITLE
Update console dialog helper documentation

### PR DESCRIPTION
**Pull request by atailouloute**
`If you reach this max number it will use the default value.` is not always true, on the last attempt we can supply an answer and the default one is not used

Actually the default answer is used each time the user is asked the question if the user didn't specify any answer and it's validated using the `validator` callback

### CODE REVIEW

**Review by roman-1983 — APPROVED** [review]

File: components/console/helpers/dialoghelper.rst [hunk 1]
```diff
@@ -142,11 +142,14 @@ in the console, so it is a good practice to put some useful information in it. T
 function should also return the value of the user's input if the validation was successful.
 
 You can set the max number of times to ask in the ``$attempts`` argument.
-If you reach this max number it will use the default value.
 Using ``false`` means the amount of attempts is infinite.
 The user will be asked as long as they provide an invalid answer and will only
 be able to proceed if their input is valid.
 
+Each time the user is asked the question, if no answer is supplied, the default one is used
+and it's validated using the ``$validator`` callback.
```
Comment (HeahDude) [inline components/console/helpers/dialoghelper.rst]: Hmm this change is legit, but not enough explicit, what about:
```rst
and it's validated using the ``$validator`` callback, printing the error
on remaining attempts. If the last attempt is reached an exception is
thrown and displayed to the user, ending the process.
```
?

Comment (javiereguiluz) [inline components/console/helpers/dialoghelper.rst]: I've reworded as suggested. This is now ready to be merged. Thanks!

**Review by javiereguiluz — APPROVED** [review]

File: components/console/helpers/dialoghelper.rst [hunk 2]
```diff
@@ -142,11 +142,15 @@ in the console, so it is a good practice to put some useful information in it. T
...
 
 You can set the max number of times to ask in the ``$attempts`` argument.
-If you reach this max number it will use the default value.
 Using ``false`` means the amount of attempts is infinite.
 The user will be asked as long as they provide an invalid answer and will only
 be able to proceed if their input is valid.
 
+Each time the user is asked the question, if no answer is supplied, the default
+one is used (and validated with the ``$validator`` callback) and the remaining
+number of attempts is displayed. If the last attempt is reached, the application
```
Comment (HeahDude) [inline components/console/helpers/dialoghelper.rst]: Hmm, "the remaining number of attempts is displayed" sounds wrong, only an error message is displayed.

**Comment by xabbuh** [comment]
Thank you @atailouloute.

**Pull request merged**